### PR TITLE
Resolving assets based on requested platform [RFC]

### LINKED ISSUE #499: AssetResolution not picking up .android assets
We've recently moved to haul and are noticing that iOS/default icons are being bundled with Android. Below is a gist containing a test I added for AssetResolution, which is the scenario that is failing for us.

https://gist.github.com/sweggersen/e1cd86575c537a91eea7c28533ecc948
(Updated to include test results)

### Your Environment

| software         | version
| ---------------- | -------
| Haul             |  v1.0.0-rc.10
| react-native     | 0.57.1
| node             | v8.11.3
| npm or yarn      | 1.10.1

### PR DESCRIPTION
**Pull request by sweggersen**
Fixes #499 .

### CODE REVIEW

**Comment by sweggersen** [comment]
FYI. This is not RFR

**Comment by sweggersen** [comment]
Updated: RFC

File: src/resolvers/__tests__/resolvers.test.js [hunk 1]
```diff
@@ -70,6 +70,47 @@ test('AssetResolver.collect returns empty object when file not in the list', ()
...
+    'mail@3x.android.png',
+    'mail@3x.png',
+  ];
+  const fileListDifferentOrder = [
+    'mail.png',
+    'mail.android.png',
+    'mail@2x.png',
+    'mail@2x.android.png',
+    'mail@3x.png',
+    'mail@3x.android.png',
```
Comment (thymikee) [inline src/resolvers/__tests__/resolvers.test.js]: can you add more platforms? some `.native.png` and `.ios.png` files to make sure this works correctly

**Comment by sweggersen** [comment]
@thymikee @acoates-ms

**Pull request merged**